core/templates/dev/head/services/SearchServiceSpec.ts

```diff
@@ -312,25 +272,25 @@ describe('Search service', function() {
       };
       var urlComponent = '?q=protractor%20test&category=("Mathematics")' +
                          '&language_code="en" OR "ar")';
-      expect(function() {
-        SearchService.updateSearchFieldsBasedOnUrlQuery(urlComponent, results);
-      }).toThrow(new Error('Invalid search query url fragment for ' +
-                           'languageCodes: language_code="en" OR "ar")'));
+      SearchService.updateSearchFieldsBasedOnUrlQuery(urlComponent, results);
       expect(results.languageCodes.selections).toEqual({});
       expect(results.categories.selections).toEqual({
         Mathematics: true
       });
 
       var urlComponent = '?q=protractor%20test&category=("Mathematics")' +
                          '&language_code="en" OR "ar"';
-      expect(function() {
-        SearchService.updateSearchFieldsBasedOnUrlQuery(urlComponent, results);
-      }).toThrow(new Error('Invalid search query url fragment for ' +
-                           'languageCodes: language_code="en" OR "ar"'));
+      SearchService.updateSearchFieldsBasedOnUrlQuery(urlComponent, results);
       expect(results.languageCodes.selections).toEqual({});
       expect(results.categories.selections).toEqual({
         Mathematics: true
       });
+
+      var urlComponent = '?q=protractor%20test&category="Mathematics"' +
```

Comment:
Perhaps put a comment above each of these blocks to explain what's being tested here (since it's not entirely obvious to the casual reader why this string is invalid).

Reply:
Done.